Fandom Creator/JavaScript.md

```diff
@@ -70,3 +70,15 @@ import './BaseCard.scss';
 ```
 
 Note that alphebatizing is based on the import directory, then file name.
+
+
+## React Components
+
+[Proposal](https://github.com/Wikia/guidelines/pull/1360)
+
+Our goal when building react components should be to make them small, composable, and reusable. 
+
+* **SHOULD** [Use Composition over Inheritance](https://reactjs.org/docs/composition-vs-inheritance.html)
+  * **SHOULD** Use [hocs](https://reactjs.org/docs/higher-order-components.html) and [render props](https://reactjs.org/docs/render-props.html) to create small reusable components. 
+* **SHOULD** avoid refs [React Docs](https://reactjs.org/docs/refs-and-the-dom.html#dont-overuse-refs)
+* **MUST** Use [function components](https://reactjs.org/docs/components-and-props.html#function-and-class-components) when possible
```

Comment:
This won't always be possible.

Reply:
Yep, should probably be *SHOULD* (which implies when possible)

Reply:
`Should....when possible` is not the same as `Must...when possible`. I don't know how to state that it is a must in the technical situation that makes it possible i.e. the only reason _not_ to do it is when you cannot.

Reply:
It's similar to the must in the binding proposal. In the situation where binding is required it MUST be done. Not always 

Reply:
We should be using these words as described in https://www.ietf.org/rfc/rfc2119.txt
In my opinion "MUST" for us means if it's not done, it would never pass code review and never be merged to master. Anything with an exception is a "SHOULD". This is just my interpretation however.

Reply:
I'm in favor of having the terms "MUST" and "SHOULD" defined explicitly, and using the RFC @mklucsarits pasted looks good to me. I created a PR to add it to the guidelines https://github.com/Wikia/guidelines/pull/138